core/src/main/java/com/kin/ecosystem/core/data/auth/AuthLocalData.java

```diff
@@ -92,24 +92,8 @@ public void run() {
 	}
 
 	@Override
-	public void getAppId(@NonNull final Callback<String, Void> callback) {
-		final Runnable command = new Runnable() {
-			@Override
-			public void run() {
-				final String appID = signInSharedPreferences.getString(APP_ID_KEY, null);
-				executorsUtil.mainThread().execute(new Runnable() {
-					@Override
-					public void run() {
-						if (appID != null) {
-							callback.onResponse(appID);
-						} else {
-							callback.onFailure(null);
-						}
-					}
-				});
-			}
-		};
-		executorsUtil.diskIO().execute(command);
+	public String getAppId() {
+		return signInSharedPreferences.getString(APP_ID_KEY, null);
```

Comment:
need to also change how raw APP_ID_KEY is saved currently it commit with IO thread - it should be just applied on UI thread (need to change it 2 methods)